libs/ipc/inc/public/tfc/ipc.hpp

```diff
@@ -30,49 +30,45 @@ auto constexpr register_cb(std::string const& ipc_name) {
  * addition with implementing the ipc connection.
  * @tparam type_desc The type description for the slot.
  */
-template <typename type_desc>
+template <typename type_desc, typename manager_client_type>
 class slot {
 public:
   using value_t = details::slot_callback<type_desc>::value_t;
   static constexpr std::string_view type_name{ type_desc::type_name };
   static auto constexpr direction_v = details::slot_callback<type_desc>::direction_v;
-
   /**
    * C'tor for a tfc IPC slot.
    * @param ctx Execution context
+   * @param client manager_client_type a reference to a manager client
    * @param name The slot name
    * @param callback Channel for value updates from the corosponding signal.
    */
-  slot(asio::io_context& ctx, std::string_view name, auto&& callback)
-      : slot_(details::slot_callback<type_desc>::create(ctx, name)), client_(ctx) {
+  slot(asio::io_context& ctx, manager_client_type& client, std::string_view name, auto&& callback)
+      : slot_(details::slot_callback<type_desc>::create(ctx, name)), client_(client) {
     client_.register_connection_change_callback(
-        [this, callback](const std::string_view slot_name, const std::string_view signal_name) {
-          if (slot_name == slot_->name_w_type()) {
-            slot_->init(signal_name, callback);
-          }
-        });
+        slot_->name_w_type(), [this, callback](const std::string_view signal_name) { slot_->init(signal_name, callback); });
     client_.register_slot(slot_->name_w_type(), type_desc::value_e, details::register_cb(slot_->name_w_type()));
   }
 
-  slot(slot&) noexcept = default;
-  slot(slot&&) noexcept = default;
-  auto operator=(slot&&) noexcept -> slot& = default;
-  auto operator=(slot const&) noexcept -> slot& = default;
+  slot(slot&) = delete;
+  slot(slot&&) noexcept = delete;
+  auto operator=(slot&&) noexcept -> slot& = delete;
+  auto operator=(slot const&) -> slot& = delete;
 
   [[nodiscard]] auto value() const noexcept { return slot_->get(); }  // todo: value() or get()
 
 private:
   static constexpr std::string_view self_name{ slot_tag };
   std::shared_ptr<details::slot_callback<type_desc>> slot_;
-  ipc_ruler::ipc_manager_client client_;
+  manager_client_type& client_;
```

Comment:
I would prefer sharing ownership of client with smart pointer, as this can result in undefined behaviour.

Reply:
I don't quite agree, using smart pointers can also result in undefined behavior. If using smart pointers we are enforcing the usage of the component forcing them to create smart pointers.

If the owner of the object wishes to use smart pointers to manage the lifetime of ipc_manager_client when using a reference he can also use a smart pointer. he is just not forced to.

f.e. 
```cpp
#include <iostream>
#include <string>
#include <memory>

struct foo{
  void scream(){
    std::cout << "AHHH" << std::endl;
  }
};

void bar(foo& f){
  f.scream();
}
int main(){
  std::shared_ptr<foo> tee = std::make_shared<foo>();
  bar(*tee);
}
```